sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/ParameterAssignedTo.cs

```diff
@@ -72,13 +65,68 @@ protected override bool IsAssignmentToParameter(ISymbol symbol)
         {
             var parameterSymbol = symbol as IParameterSymbol;
             var result = parameterSymbol?.RefKind == RefKind.None;
-
             return result;
         }
 
+        protected override bool IsReadBefore(SemanticModel semanticModel, ISymbol symbol, AssignmentExpressionSyntax assignment)
+        {
+            return GetPreviousStatements(assignment).Union(new[] { assignment.Right })

```

Comment:
[cosmetic] Please move the `Union` statement to the next line. It's easier to read chained elements.